### DISCUSSION TITLE
Custom link jinja2 rendering with namespaces

### DISCUSSION THREAD

**Discussion by florianschendel** [netbox-community/netbox#12524] [Help Wanted!]
Hi,

since version netbox 3.5 we have problems with our existing "custom link" jinja2 rendering.
It looks like that namespaces are not longer supported.

e.g.
```
set myVariable = namespace(value=object.cf.myCustomFiledValue %}
```

Any ideas?

**Comment by abhi1693** [comment]
That does not seem like a valid syntax.

**Comment by bitcollector1** [comment]
We also have a custom link that broke with the upgrade, several of them actually but I was able to fix the rest, mostly related to object. 

However, this one refuses to work correctly with the new 3.5 
```
{%if 'minios' in object.device_role.name%} 
{%set link = "discovered_hosts"%}
{%elif 'server' in object.device_role.name%} 
{%set link = "hosts"%} 
{%endif%}
{%set int_name =  object.primary_ip.assigned_object.name%}
{%set mac = object.interfaces.get(name=int_name).mac_address%}
https://spc01.hwe.ebay.com/{{link}}?utf8=&search=mac={{mac}}
```

Renders a link that no longer works 

https://netdevbox.hwe.ebay.com/dcim/devices/1973/+%0A+++++%0A%0A%0A%0Ahttps://spc01.hwe.ebay.com/hosts?utf8=&search=mac=AC:1F:6B:09:A4:9C

Link should be:
https://spc01.hwe.ebay.com/hosts?utf8=&search=mac=AC:1F:6B:09:A4:9C

**Reply by kkthxbye-code** [reply]
Here you go:

```
{%-if 'minios' in object.device_role.name%} 
{%-set link = "discovered_hosts"%}
{%-elif 'server' in object.device_role.name%} 
{%-set link = "hosts"%} 
{%-endif%}
{%-set int_name =  object.primary_ip.assigned_object.name%}
{%-set mac = object.interfaces.get(name=int_name).mac_address-%}
https://spc01.hwe.ebay.com/{{link}}?utf8=&search=mac={{mac}}
```

I think the patch here https://github.com/netbox-community/netbox/issues/12205 might have changed the way whitespace is handled.

**Reply by bitcollector1** [reply]
Very kind of you to offer the solution to my issue, thanks very much!

**Comment by bitcollector1** [comment]
just to confirm, it works perfectly :)
